app/models/katello/events/delete_latest_content_view_version.rb

```diff
@@ -0,0 +1,40 @@
+module Katello
+  module Events
+    class DeleteLatestContentViewVersion
+      EVENT_TYPE = 'delete_latest_content_view_version'.freeze
+
+      attr_reader :content_view
+      attr_accessor :metadata, :retry
+
+      def self.retry_seconds
+        180
+      end
+
+      def initialize(content_view_id)
+        @content_view = ::Katello::ContentView.find_by_id(content_view_id)
+        Rails.logger.warn "Content View not found for ID #{object_id}" if @content_view.nil?
+        yield(self) if block_given?
+      end
+
+      def run
+        return unless content_view
+
+        begin
+          ForemanTasks.async_task(::Actions::Katello::ContentView::Remove, content_view,
```

Comment:
This is where the actual removal happens.

Reply:
Is there ever a chance the cv can be nil?